typescript/agentkit/src/wallet-providers/privyEvmEmbeddedWalletProvider.ts

```diff
@@ -0,0 +1,517 @@
+import { Network } from "../network";
+import axios from "axios";
+import canonicalize from "canonicalize";
+import crypto from "crypto";
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import {
+  TransactionRequest,
+  PublicClient,
+  Abi,
+  ReadContractParameters,
+  ReadContractReturnType,
+  ContractFunctionArgs,
+  ContractFunctionName,
+  TransactionReceipt,
+  parseEther,
+  Address,
+  Hex,
+} from "viem";
+import { getChain, NETWORK_ID_TO_CHAIN_ID } from "../network/network";
+import { PrivyWalletConfig, PrivyWalletExport, createPrivyClient } from "./privyShared";
+import { createPublicClient, http } from "viem";
+import { WalletWithMetadata } from "@privy-io/server-auth";
+import { WalletProvider } from "./walletProvider";
+
+interface PrivyResponse<T> {
+  data: T;
+}
+
+/**
+ * Configuration options for the Privy Embedded Wallet provider.
+ */
+export interface PrivyEvmEmbeddedWalletConfig extends PrivyWalletConfig {
+  /** The ID of the delegated wallet */
+  walletId: string;
+
+  /** The network ID to connect to (e.g., "base-mainnet") */
+  networkId?: string;
+
+  /** The chain ID to connect to */
+  chainId?: string;
+}
+
+/**
+ * A wallet provider that uses Privy's embedded wallets with delegation.
+ * This provider extends the EvmWalletProvider to provide Privy-specific wallet functionality
+ * while maintaining compatibility with the base wallet provider interface.
+ */
+export class PrivyEvmEmbeddedWalletProvider extends WalletProvider {
+  #walletId: string;
+  #address: string;
+  #appId: string;
+  #appSecret: string;
+  #authKey: string;
+  #network: Network;
+  #publicClient: PublicClient;
+
+  /**
+   * Private constructor to enforce use of factory method.
+   *
+   * @param config - The configuration options for the wallet provider
+   */
+  private constructor(config: PrivyEvmEmbeddedWalletConfig & { address: string }) {
+    super();
+
+    this.#walletId = config.walletId;
+    this.#address = config.address;
+    this.#appId = config.appId;
+    this.#appSecret = config.appSecret;
+    this.#authKey = config.authorizationPrivateKey || "";
+
+    const networkId = config.networkId || "base-sepolia";
+    const chainId = config.chainId || NETWORK_ID_TO_CHAIN_ID["base-sepolia"];
```

Comment:
Ran into a bug while testing. Please update this to:

```ts
const chainId = config.chainId || NETWORK_ID_TO_CHAIN_ID[networkId];
```

So that the fallback behavior of the chainId is to derive it from the network

Reply:
makes sense. updated